src/geode/geometry/distance.cpp

```diff
@@ -586,6 +586,80 @@ namespace geode
         return std::make_tuple( 0, point );
     }
 
+    std::tuple< double, Point3D > point_circle_distance(
+        const Point3D& point, const Circle& circle )
+    {
+        const Vector3D center_to_point{ circle.plane().origin(), point };
+        const auto distance_to_plane =
+            center_to_point.dot( circle.plane().normal() );
+        const auto projected_on_plane =
+            point - circle.plane().normal() * distance_to_plane;
+        const Vector3D center_to_projected_point{ circle.plane().origin(),
+            projected_on_plane };
+        if( center_to_projected_point.length() < global_epsilon )
+        {
+            Vector3D other_direction{ { 1.0, 0.0, 0.0 } };
+            if( circle.plane().normal().inexact_equal(
+                    other_direction, global_epsilon )
+                || circle.plane().normal().inexact_equal( other_direction * -1.,
+                    global_epsilon ) ) // dummy_direction is aligned along plane
+                                       // normal
+            {
+                other_direction.set_value( 1, 1.0 );
+            }
+            OPENGEODE_ASSERT( !circle.plane().normal().inexact_equal(
+                                  other_direction, global_epsilon ),
+                "[point_circle_distance] Problem while getting circle nearest "
+                "point" );
+            const Vector3D other_projected_on_plane =
```

Comment:
```suggestion
            const auto other_projected_on_plane =
```
Is it working?

Reply:
No, auto stands for Point3D that cannot be normalized()